### PR TITLE
Handle beam tables with >1 stokes component: fix for #825

### LINKED ISSUE #825: Stokes Cube with beams in multiple axis can't be read in. Prevents the loading any cleaned CASA IQUV image.
`cube = SpectralCube.read(filenameL, format='casa', use_dask=True)`
or 
`cube = StokesSpectralCube.read(filenameL, format='casa', use_dask=True)`

results in the same error.
File ~/miniconda3/envs/py38/lib/python3.8/site-packages/spectral_cube/io/casa_image.py:109, in load_casa_image(filename, skipdata, memmap, skipvalid, skipcs, target_cls, use_dask, target_chunksize, **kwargs)
    107 bdict = beam_['beams']
    108 if beam_['nStokes'] > 1:
--> 109     raise NotImplementedError()
    110 nbeams = len(bdict)
    111 assert nbeams == beam_['nChannels']

NotImplementedError:

### PR DESCRIPTION
**Pull request by e-koch**
This PR implements fixes #825.

* [x] Fix for FITS files, where the beam table is not split into the different components
* [x] Fix for CASA images where we don't have reading beams for different components implemented.
* [x] Tests for reading/writing Stokes cubes with beam tables

### CODE REVIEW

**Comment by keflavich** [comment]
LGTM.  A test would probably be good to add.

Can we merge this with just FITS and move the 'todo' list to issue #825 (and reopen it)?

**Comment by e-koch** [comment]
I'd like to take a look on how much work the CASA image reader. It _may_ not be too bad and, if so, would be nice to clear off in this PR. I'll update here in a few days.

**Comment by e-koch** [comment]
The CASA reader fix wasn't too bad.

In both cases we'll still hit issues when the components aren't IQUV, but that's a separate issue.

**Comment by e-koch** [comment]
Eventually we may want to move some of this into radio-beam for the polarization handling (especially any masking operations across polarizations).

File: spectral_cube/tests/test_casafuncs.py [hunk 1]
```diff
@@ -68,6 +68,43 @@ def make_casa_testimage(infile, outname):
...
         ia.done()
 
+def make_casa_stokes_testimage(infile, outname):
+
+    infile = str(infile)
+    outname = str(outname)
+
+    if not CASA_INSTALLED:
+        raise Exception("Attempted to make a CASA test image in a non-CASA "
+                        "environment")
```
Comment (keflavich) [inline spectral_cube/tests/test_casafuncs.py:78]: do we not have a decorator for this?

**Review by keflavich — APPROVED** [review]

File: spectral_cube/tests/test_casafuncs.py (hunk 1 above)
Comment (e-koch) [reply spectral_cube/tests/test_casafuncs.py:78]: I don't think we have one for the tests.

**Pull request merged**